plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -549,12 +549,11 @@ private static TableIdentifier toIdentifier(SchemaTableName schemaTableName)
 
     private List<Namespace> listNamespaces(ConnectorSession session, Optional<String> namespace)
     {
-        if (namespace.isEmpty()) {
-            return listNamespaces(session).stream()
-                    .map(Namespace::of)
-                    .collect(toImmutableList());
-        }
-
-        return ImmutableList.of(Namespace.of(namespace.get()));
+        return namespace
```

Comment:
This seems significantly harder to read to me compared to the previous version with the conditional

Reply:
It's a matter of taste, really. I prefer the functional style, personally. I reverted all changes from this commit except in the file where it's needed for further changes (they were also requested by a maintainer).